bridge/setu/processor/staking.go

```diff
@@ -333,3 +385,61 @@ func (sp *StakingProcessor) isOldTx(cliCtx cliContext.CLIContext, txHash string,
 
 	return status, nil
 }
+
+func (sp *StakingProcessor) checkValidNonce(validatorId uint64, txnNonce uint64) (bool, uint64, error) {
+	currentNonce, currentHeight, err := util.GetValidatorNonce(sp.cliCtx, validatorId)
+	if err != nil {
+		sp.Logger.Error("Failed to fetch validator nonce and height data from API", "validatorId", validatorId)
+		return false, 0, err
+	}
+
+	if currentNonce+1 != txnNonce {
+		sp.Logger.Error("Nonce for the given event not in order", "validatorId", validatorId, "currentNonce", currentNonce, "txnNonce", txnNonce)
+		return false, txnNonce - currentNonce, nil
+	}
+
+	stakingTxnCount, err := queryTxCount(sp.cliCtx, validatorId, currentHeight)
+	if err != nil {
+		sp.Logger.Error("Failed to query stake txns by txquery for the given validator", "validatorId", validatorId)
+		return false, 0, err
+	}
+
+	if stakingTxnCount != 0 {
+		sp.Logger.Info("Recent staking txn count for the given validator is not zero", "validatorId", validatorId, "currentNonce", currentNonce, "txnNonce", txnNonce, "currentHeight", currentHeight)
+		return false, 1, nil
```

Comment:
@0xKrishna , why are we returning 1 here, sorry if I forgot this from our last discussion.

Reply:
To make the incoming wait for 1 unit of time (i.e. 10 Seconds).
Why 1 unit only? Because if the txn has passed the nonce check that means that it's the next txn in line and we should minimize the wait in which it will come back again.